apis/python/src/tiledb/vector_search/flat_index.py

```diff
@@ -136,8 +136,20 @@ def query_internal(
                 (queries.shape[0], k), MAX_UINT64
             )
 
+        try:
+            distance_metric = vspy.DistanceMetric(
+                self.group.meta.get("distance_metric", vspy.DistanceMetric.L2)
+            )
+        except ValueError:
+            distance_metric = vspy.DistanceMetric.L2
+            raise ValueError(
```

Comment:
If you want to fallback to L2 as default you shouldn't use raise here.

Also `distance_metric` should only be configured during the index constructor.

Reply:
Removed the fallback as it makes better sense to raise ther